airflow/api_connexion/endpoints/base.py

```diff
@@ -0,0 +1,242 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, Container, Dict, Generic, Iterable, List, Optional, Tuple, Type, TypeVar
+
+from flask import request
+from marshmallow import Schema, ValidationError
+from sqlalchemy.orm import Query
+from sqlalchemy.sql import ColumnElement
+
+from airflow import settings
+from airflow._vendor.connexion import NoContent
+from airflow.api_connexion.exceptions import AlreadyExists, BadRequest, NotFound
+from airflow.api_connexion.parameters import check_limit, format_parameters
+from airflow.compat.functools import cached_property
+from airflow.models.base import Base
+
+M = TypeVar("M", bound=Base)
+
+
+class Endpoints(Generic[M]):
+    """Base endpoint implementation generator class for an abstract resource.
+
+    This class does not assume the data storage backend, and a resource must
+    implement various methods to make things work. If the resource is a
+    SQLAlchemy model, see ``ModelEndpoints``.
+    """
+
+    resource_name: str  # Name of the resource for display in error messages etc.
+    instance_index: List[str]  # List keys to uniquely select an instance.
```

Comment:
```suggestion
    instance_indices: List[str]  # List keys to uniquely select an instance.
```
Wondering if it should be plural?

Reply:
This is not a list of indices, but describing _an_ index (in the database sense) to select a unique instance out for comparison (another name I briefly considered was `primary_key`). But the name is indeed difficult to understand, maybe I'll change it to something like `unique_instance_fields`. Suggestions are definitely much welcomed.

Reply:
Cool. I prefer the  `unique_instance_fields`